iOS Enrollment integration test fails to build: untrusted Swift macro (ModifiedCopy)

## Problem

The **iOS Enrollment** integration test fails at the "Build Fennec scheme" step (the enrollment test itself never runs). Surfaced on the automated iOS-SHA bump PR #15894.

Build error:

```
ModifiedCopyMacro/Package.swift: error: Macro "ModifiedCopyMacros" from package "ModifiedCopy" must be enabled before it can be used
** BUILD FAILED **  (exit code 65)
```

A newer firefox-ios `main` pulled in a dependency on the `ModifiedCopy` Swift macro package. Since Swift 5.9, third-party macros run arbitrary code at compile time and must be explicitly trusted before they compile — interactive Xcode shows a "Trust & Enable" prompt, but non-interactive `xcodebuild` in CI just fails. This is a recurring class: every new compile-time macro package in firefox-ios will fail the auto-bump PR the same way until CI is told to trust macros.

## Fix

Add `-skipMacroValidation` to the `xcodebuild build` invocation in `.github/workflows/ios-integration-test.yml`. This bypasses the trust check on macros from the firefox-ios source we just checked out; it has no effect on what the enrollment test validates.

## Acceptance criteria

- The `xcodebuild build` step in the iOS integration workflow skips macro fingerprint validation.
- The iOS Enrollment integration test builds Fennec and runs the enrollment test to completion.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-6982)

--- a/.github/workflows/ios-integration-test.yml
+++ b/.github/workflows/ios-integration-test.yml
@@ -121,6 +121,7 @@ jobs:
             -scheme Fennec \
             -destination "platform=iOS Simulator,name=${IOS_SIMULATOR},OS=${IOS_VERSION}" \
             -derivedDataPath ./DerivedData \
+            -skipMacroValidation \
             COMPILER_INDEX_STORE_ENABLE=NO \
             CODE_SIGN_IDENTITY= CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO \
             ENABLE_USER_SCRIPT_SANDBOXING=NO \

